browser/js/dashboard/gridbox/widgetCtrl.js

```diff
@@ -1,5 +1,7 @@
-app.controller('WidgetCtrl', ['$scope', '$uibModal',
-    function($scope, $uibModal) {
+
+app.controller('WidgetCtrl', ['$scope', '$modal', '$controller', '$rootScope', 'WidgetSettingsFactory', 'DashboardFactory', '$interval', '$uibModal',
+	function($scope, $modal, $controller, $rootScope, WidgetSettingsFactory, DashboardFactory, $interval, $uibModal) {
+
 
       $scope.remove = function(widget) {
         $scope.dashboard.charts.splice($scope.dashboard.charts.indexOf(widget), 1);
```

Comment:
Watch out for this. If the widget is not in the array you will end up splicing out the last element! [Like this](https://repl.it/CpDO).